src/main.cpp

```diff
@@ -3617,10 +3618,44 @@ void UnlinkPrunedFiles(std::set<int>& setFilesToPrune)
     }
 }
 
+/* This function is called from the RPC code for pruneblockchain */
+void PruneBlockFilesManual(int manualPruneHeight)
+{
+    // stuff the prune height into a global variable, and flush
```

Comment:
I usually get told off for using global variables. Curious to know when they are permitted and when they are discouraged.